serverHandlers/middleware.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const UserModel = require('../Schemas/UserSchema');
+const fs = require('fs');
+let path = require('path');
+const allowedImgExts = [
+    "image/jpeg",
+    "image/jpg",
+    "image/gif",
+    "image/png"
+];
+
+exports.validateInputFields = async (req, res, next) => {
+    let userInfo = req.body;
+
+    let checkEmail = userInfo.email;
+    let imgPath = req.file.path;
+
+    let imgRemove = (imgPath) => {
+        // had to use sync method, async needs to have a callback - do not see profit of it
+        fs.unlinkSync(path.join(__dirname, '..', imgPath));
+    };
+
+    UserModel.find({userEmail: checkEmail}, (err, data) => {
+        if(data.length){
+            imgRemove(imgPath);
+            res.status(400).end(JSON.stringify({message: "User already exists!"}));
+        }
+    });
+
+    // validation check
+    if(!(/^([a-zA-Z]{1,32})$/.test(userInfo.name))){
```

Comment:
Is there a way to optimize this hundreds of lines  if...else if?

Reply:
optimized and added functionality for validation